src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -589,31 +602,34 @@ private boolean retryUnsubscribe(UnsubAckPacket.UnsubAckReasonCode rc) {
 
     @Override
     public void start()  {
-        if (clientStarted) {
-            return;
+        if (isSSL()) {
+            mqttClientKeyStore.listenToUpdates(onKeyStoreUpdate);
         }
         client.start();
-        clientStarted = true;
     }
 
     @Override
     public void stop() {
-        mqttClientKeyStore.unsubscribeFromCAUpdates(onKeyStoreUpdate);
-        cancelUpdateSubscriptionsTask();
+        mqttClientKeyStore.unsubscribeFromUpdates(onKeyStoreUpdate);
+        cancelResetRetry();
         closeClient();
     }
 
     private void closeClient() {
+        cancelUpdateSubscriptionsTask();
+        doCloseClient();
+    }
+
+    private void doCloseClient() {
         try {
-            if (clientStarted) {
+            if (client.getIsConnected()) {
                 client.stop(null);
             } else {
                 client.close();
```

Comment:
why is this in the else? It needs to be closed no matter what, so just remove the else?

Reply:
stop invokes the Mqtt5ClientOptions.LifecycleEvents#onStopped handler, which calls close

Reply:
i think the reason is to give a chance for the disconnect packet to reach the broker

Reply:
Do you expect that this branch would ever happen?

Reply:
thinking through this one a bit more...

Reply:
by branch you mean this `client.close()` right? that could happen if reset() is called during a retry scenario

Reply:
The `else` branch, yes.